stream-chat-android-compose-sample/src/main/java/io/getstream/chat/android/compose/sample/ui/StartupActivity.kt

```diff
@@ -0,0 +1,56 @@
+package io.getstream.chat.android.compose.sample.ui
+
+import android.content.Context
+import android.content.Intent
+import android.os.Bundle
+import androidx.appcompat.app.AppCompatActivity
+import androidx.core.app.TaskStackBuilder
+import io.getstream.chat.android.compose.sample.ChatApp
+import io.getstream.chat.android.compose.sample.ui.login.UserLoginActivity
+
+/**
+ * An Activity without UI responsible for startup routing. It navigates the user to
+ * one of the following screens:
+ *
+ * - Login screen, if the user is not authenticated
+ * - Channels screen, if the user is authenticated
+ * - Messages screen, if the user is coming from a push notification
+ */
+class StartupActivity : AppCompatActivity() {
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+
+        val userCredentials = ChatApp.credentialsRepository.loadUserCredentials()
+        if (userCredentials != null) {
+            // Ensure that the user is connected
+            ChatApp.chatManager.connectUser(userCredentials)
```

Comment:
Should we add some callbacks here for the connection? as if something goes wrong it might cause a bad state